### PR TITLE
library added

### PR DESCRIPTION
**Pull request by RomanKondratovich**

### CODE REVIEW

**Comment by sudden6** [comment]
Which compile are you using?

**Comment by RomanKondratovich** [comment]
Hello,
Target: x86_64-w64-mingw32

Configured with: ../gcc-12.1.0/configure --prefix=/mingw64
--with-local-prefix=/mingw64/local --build=x86_64-w64-mingw32
--host=x86_64-w64-mingw32 --target=x86_64-w64-mingw32
--with-native-system-header-dir=/mingw64/include --libexecdir=/mingw64/lib
--enable-bootstrap --enable-checking=release --with-arch=x86-64
--with-tune=generic
--enable-languages=c,lto,c++,fortran,ada,objc,obj-c++,jit --enable-shared
--enable-static --enable-libatomic --enable-threads=posix --enable-graphite
--enable-fully-dynamic-string --enable-libstdcxx-filesystem-ts
--enable-libstdcxx-time --disable-libstdcxx-pch --enable-lto
--enable-libgomp --disable-multilib --disable-rpath
--disable-win32-registry --disable-nls --disable-werror --disable-symvers
--with-libiconv --with-system-zlib --with-gmp=/mingw64 --with-mpfr=/mingw64
--with-mpc=/mingw64 --with-isl=/mingw64 --with-pkgversion='Rev3, Built by
MSYS2 project' --with-bugurl=https://github.com/msys2/MINGW-packages/issues
--with-gnu-as --with-gnu-ld --disable-libstdcxx-debug
--with-boot-ldflags=-static-libstdc++
--with-stage1-ldflags=-static-libstdc++

Thread model: posix

Supported LTO compression algorithms: zlib zstd

gcc version 12.1.0 (Rev3, Built by MSYS2 project)
Roman K.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,211 @@
...
+if (${BUILD_STATIC})
+    # Linking standard libs statically
+    set(CMAKE_CXX_STANDARD_LIBRARIES "-Wl,-Bstatic,--whole-archive -lpthread -Wl,--no-whole-archive -static-libgcc -static-libstdc++")
+    set(CMAKE_EXE_LINKER_FLAGS "--static --verbose")
+    target_link_libraries(m-queens2-boinc-ocl ${BOINC_STATIC_DIR}/lib/libboinc_api.a)
+    target_link_libraries(m-queens2-boinc-ocl ${BOINC_STATIC_DIR}/lib/libboinc_opencl.a)
+    target_link_libraries(m-queens2-boinc-ocl ${BOINC_STATIC_DIR}/lib/libboinc.a)
+else()
+    target_link_libraries(m-queens2-boinc-ocl boinc)
+    target_link_libraries(m-queens2-boinc-ocl boinc_api)
```
Comment (Hagb) [inline CMakeLists.txt:208]: `boinc_opcncl` should also be linked if m-queens2-boinc-ocl is built with dynamic link.
```suggestion
    target_link_libraries(m-queens2-boinc-ocl boinc_api)
    target_link_libraries(m-queens2-boinc-ocl boinc_opencl)
```

Comment (sudden6) [reply CMakeLists.txt:208]: I guess that should also be applied to the statically linked version, right?

Comment (Hagb) [reply CMakeLists.txt:208]: > I guess that should also be applied to the statically linked version, right?

I think so. That has been applied to the statically linked version in this PR. (CMakeLists.txt 204L)

Comment (sudden6) [reply CMakeLists.txt:208]: My mistake, that's already in the code anyway.

**Comment by sudden6** [comment]
Both of your suggested changes should now be in https://github.com/sudden6/m-queens/tree/gpu_recursive thanks for the fixes!

**Pull request closed without merging**